bottles/backend/managers/dependency.py

```diff
@@ -491,32 +493,31 @@ def __step_archive_extract(self, step: dict):
             checksum=step.get("file_checksum")
         )
 
-        if download:
-            if step.get("rename"):
-                file = step.get("rename")
-            else:
-                file = step.get("file_name")
+        if not download:
+            return False
 
-            archive_path = os.path.join(Paths.temp, os.path.splitext(file)[0])
+        if step.get("rename"):
+            file = step.get("rename")
+        else:
+            file = step.get("file_name")
 
-            if os.path.exists(archive_path):
-                shutil.rmtree(archive_path)
+        archive_path = os.path.join(Paths.temp, os.path.splitext(file)[0])
 
-            os.makedirs(archive_path)
-            try:
-                ext_path = patoolib.extract_archive(os.path.join(Paths.temp, file), outdir=archive_path)
-                ext_file = ext_path + '/' + os.path.basename(ext_path)
-                if os.path.exists(archive_path):
-                    if os.path.isfile(ext_file):
-                        patoolib.extract_archive(
-                        ext_file,
-                        outdir=ext_path + '/'
-                        )
-            except:
-                return False
-            return True
+        if os.path.exists(archive_path):
+            shutil.rmtree(archive_path)
 
-        return False
+        os.makedirs(archive_path)
+        try:
+            patoolib.extract_archive(os.path.join(Paths.temp, file), outdir=archive_path)
+            if archive_path.endswith(".tar"):
+                tar_path = os.path.join(archive_path, os.path.basename(archive_path))
+                patoolib.extract_archive(tar_path, outdir=archive_path)
```

Comment:
This seems to have broken d3dx9 dependency installation. The file `d3dx9.tar.xz` is extracted into the directory `temp/d3dx9.tar/d3dx9` and nothing else need to be done (see the url definitions in [d3dcompiler_47.yml](https://github.com/bottlesdevs/dependencies/blob/main/Essentials/d3dcompiler_47.yml)). Previously there was a `if os.path.isfile(ext_file)` condition to avoid the second extracting, but now it is attempted anyways and gives an error `temp/d3dx9.tar/d3dx9.tar was not found`.